app/src/main/java/com/example/weather/screen/map/MapFragment.kt

```diff
@@ -104,6 +209,57 @@ class MapFragment private constructor() : BaseFragment<FragmentMapBinding>(), On
         mGoogleMap?.animateCamera(newLatLngZoom)
     }
 
+    private fun moveToLocation(latitude: Double, longitude: Double) {
+        mGoogleMap?.clear()
+        zoomOnMap(LatLng(latitude, longitude))
+        addMarker(LatLng(latitude, longitude))
+    }
+
+    override fun onProgressLoading(isLoading: Boolean) {
+        if (isLoading) {
+            viewBinding.progressBar.visibility = View.VISIBLE
+        } else {
+            viewBinding.progressBar.visibility = View.GONE
+        }
+    }
+
+    override fun onGetCurrentWeatherSuccess(weather: Weather) {
+        mWeather = weather
+        mWeather?.let { bindDataToView(it) }
+    }
+
+    @Suppress("NestedBlockDepth")
+    private fun bindDataToView(weather: Weather) {
+        weather.weatherCurrent?.let { weatherCurrent ->
+            val time = weatherCurrent.dateTime?.unixTimestampToHourString()?.toInt()
+            if (time != null) {
+                weatherCurrent.weatherMainCondition?.let { mainCondition ->
+                    getIcon(mainCondition, time)?.let { image ->
+                        viewBinding.layoutWeatherMap.ivMainCondition.setImageResource(image)
+                    }
+                }
+            }
+            viewBinding.layoutWeatherMap.tvMainCondition.text = weatherCurrent.weatherMainCondition
+            viewBinding.layoutWeatherMap.tvTemperature.text = weatherCurrent.temperature?.kelvinToCelsius().toString()
+        }
+        viewBinding.layoutWeatherMap.tvLocation.text = weather.city + ", " + weather.country
```

Comment:
```suggestion
        viewBinding.layoutWeatherMap.tvLocation.text = weather.getLocation()
```
nên handle data trả về bên trong model